src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -592,21 +575,65 @@ public boolean addMissingAlleles(final Collection<A> candidateAlleles, final dou
         return true;
     }
 
+    /**
+     * Group evidence into lists of evidence -- for example group by read name to force read pairs to support a single haplotype
+     * @return
+     */
+    public <U> AlleleLikelihoods<GroupedEvidence<EVIDENCE>, A> groupEvidence(final Function<EVIDENCE, U> groupingFunction) {
```

Comment:
Don't see where you use this ```GroupedEvidence<?>``` objects perhaps there is staff missing in the PR? 

It seems that for you current use what you  would like to do is to return a Fragment, and perhaps some other people may want to return something else. You don't need to commit to a particular class for the return' evidence but instead you can allow the user of this method to provide its own ```Collector``` to produce this objects to fit their needs. 

I guess the easiest at this point is to provide to define a lambda ```Function<List<EVIDENCE>, NEW_EVIDENCE_TYPE>```. .... You can get some insipiration from Stream's Collectors.groupingBy ... the one that accept a list mapping lambda (a Collector in fact) ....

Reply:
done